src/coreclr/nativeaot/System.Private.CoreLib/src/System/Reflection/ConstructorInvoker.cs

```diff
@@ -40,43 +43,67 @@ public object Invoke()
         [DebuggerGuidedStepThrough]
         public object Invoke(object? arg1)
         {
-            object result = _methodBaseInvoker.CreateInstanceWithFewArgs(new Span<object?>(ref arg1));
+            if (_parameterCount > 1)
```

Comment:
I would expect that the argument count needs to match exactly. Why would we want to enable passing of unused arguments?

Reply:
The caller may special case <=4 args and have a single call to invoke like as is done in this PR: https://github.com/dotnet/runtime/pull/90119/files#diff-fb8d7b29b1ef5b84f9327e76355798f150b03d46d2e2595c561df1b060ca4ca7R755

I'm OK with enforcing exact match however (we could always loosen it in the future).